lib/esp32.toit

```diff
@@ -100,3 +100,19 @@ image_config_ -> ByteArray:
 
 set_real_time_clock_ seconds/int ns/int -> none:
   #primitive.core.set_real_time_clock
+
+/**
+Sets the RTC user data to the given $data starting from the given $offset.
+
+The given $data must be within bounds. That is 0 <= $offset <= $offset + data.size < 2048.
+*/
+set_rtc_data data offset/int=0 -> none:
```

Comment:
Depending on how nice we want to make the RTC api, we could provide a singleton with `operator[]` ..., but as an underlying API this is probably good enough.

Reply:
We could definitely do that, but we should probably add a primitive specifically for that to reduce the overhead.

Reply:
Maybe the correct thing to do is to make a `ByteArray` with an external allocation that points to the RTC memory.